.github/workflows/docs.yml

```diff
@@ -0,0 +1,47 @@
+
+name: Documentation
+on:
+  push:
+    branches: [ master ]
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+        
+      - name: Prepare apt repositories
+        run: | 
+          sudo sed -i '/deb-src/s/^# //' /etc/apt/sources.list
+          sudo apt update
+
+      - name: Install sugar-toolkit-gtk3
+        run: |
+          # enable deb-src repositories
+          
+          sudo apt build-dep sugar-toolkit-gtk3
+          sudo apt install python-is-python3 python3-cairo python3-gi python3-gi-cairo python3-dateutil python3-decorator \
+            gir1.2-telepathyglib-0.12 gir1.2-webkit2-4.0 python3-empy gir1.2-sugarext-1.0
```

Comment:
Shouldn't `apt build-dep sugar-toolkit-gtk3` install everything we need to build from sources? Then why do we need to install these additional packages manually?

I would use either one method or the other, but not both. Since the build deps of sugar-toolkit-gtk3 will likely drift from what was included with the build machine (ubuntu 20.04), listing the deps explicitly seems better.

Reply:
We do not have a list of dependencies used by sugar. There were attempts to create one, but it never worked. See https://github.com/sugarlabs/sugar/issues/884#issuecomment-561384226 for some @quozl's comment during my last GCI. (reading that thread makes me regret for being dumb that time). 

We need to install the additional packages, because `build-dep` only fetches the build time dependency. To run `sugar-toolkit-gtk3`, we need to get some `runtime` dependencies. These runtime dependencies will be used to generate links to other API documentations like [python-gi documentation](https://lazka.github.io/pgi-docs) iirc. But, that becomes some extra work to keep the build time deps updated I guess :smile:. I am aware of all build time and runtime dependencies used on [Archlinux](https://aur.archlinux.org/packages/sugar-toolkit-gtk3-git/), might be helpful.

Reply:
`build-dep` doesn't do the runtime dependencies.  Also we tend to be ahead of the dependencies here, so relying the current Ubuntu package (which will likely be the latest Debian Testing package) isn't the best way.

Placing a list of build and runtime dependencies in the repository could be more discoverable; I don't know what other projects use, but perhaps `debian-build-dependencies` and `debian-runtime-dependencies` might be okay.

Reply:
> `build-dep` doesn't do the runtime dependencies. Also we tend to be ahead of the dependencies here, so relying the current Ubuntu package (which will likely be the latest Debian Testing package) isn't the best way.
> 
> Placing a list of build and runtime dependencies in the repository could be more discoverable; I don't know what other projects use, but perhaps `debian-build-dependencies` and `debian-runtime-dependencies` might be okay.

Agreed, placing that in the repository would solve the problem. Alternatively, using Arch Linux to build the packages would also solve the problem, as we always build on the latest stuff, and deps are always upto date. We can do either of them.

Reply:
Up to you what to use.  I'm not expecting to maintain this; you've demonstrated the skills and capacity, so I'll leave it in your hands.  That said, if we do have dependency lists in the source repository, we might also change our documentation to refer to them instead of using [the `build-dep` hack](https://github.com/sugarlabs/sugar/blob/master/docs/development-environment.md#native-sugar).